app/controllers/books_controller.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+class BooksController < ApplicationController
+  before_action :authenticate_user!, :set_book, except: %i[index new create]
+
+  def index
+    @ransack = Book.ransack(params[:q])
```

Comment:
why `@ransack` is instance variable? it's not called anywhere from the respective view